Lib/test/test_xml_etree.py

```diff
@@ -2680,18 +2681,101 @@ class Y(X, ET.Element):
         e = ET.Element('foo')
         e.extend(L)
 
-    def test_remove_with_mutating(self):
-        class X(ET.Element):
+    def test_remove_with_clear(self):
+        # See: https://github.com/python/cpython/issues/126033
+
+        # Until the discrepency between "del root[:]" and "root.clear()" is
+        # resolved, we need to keep two tests. Previously, using "del root[:]"
+        # did not crash with the reproducer of gh-126033 while "root.clear()"
+        # did.
+
+        E = ET.Element
+
+        def test_remove(root, target, raises):
+            if raises:
+                self.assertRaises(ValueError, root.remove, target)
+            else:
+                root.remove(target)
+                self.assertNotIn(target, root)
+
+        for raises in [True, False]:
+
+            class X(E):
+                def __eq__(self, o):
+                    del root[:]
+                    return not raises
+
+            class Y(E):
+                def __eq__(self, o):
+                    root.clear()
+                    return not raises
+
+            for etype, rem_type in [(E, X), (X, E), (E, Y), (Y, E)]:
+                with self.subTest(
+                    etype=etype, rem_type=rem_type, raises=raises,
+                ):
+                    with self.subTest('single child'):
+                        root = E('.')
+                        root.append(etype('one'))
+                        test_remove(root, rem_type('missing'), raises)
+
+                    with self.subTest('with children'):
+                        root = E('.')
+                        root.extend([etype('one'), rem_type('two')])
+                        test_remove(root, rem_type('missing'), raises)
```

Comment:
I am still not sure that this test is needed. In any case, it tests removing `X()` from `[E(), X()]` and removing `E()` from `[X(), E()]`. The type of the second children is irrelevant, it should be E for larger symmetry.
```suggestion
                        root.extend([etype('one'), E('two')])
                        test_remove(root, rem_type('missing'), raises)
```